fedimint-client/src/sm/state.rs

```diff
@@ -6,17 +6,14 @@ use std::sync::Arc;
 
 use fedimint_core::core::ModuleInstanceId;
 use fedimint_core::db::DatabaseTransaction;
+use fedimint_core::dyn_newtype_define_with_instance_id;
 use fedimint_core::encoding::{Decodable, DynEncodable, Encodable};
-use fedimint_core::{
-    dyn_newtype_define_with_instance_id, dyn_newtype_impl_dyn_clone_passhthrough_with_instance_id,
-    module_dyn_newtype_impl_encode_decode, newtype_impl_eq_passthrough_with_instance_id,
-};
 use futures::future::BoxFuture;
 
 use crate::sm::{GlobalContext, OperationId};
 
 /// Implementors act as state machines that can be executed
-pub trait State:
+pub trait State<GC>:
```

Comment:
I'm not 100% sure if it should be an associated type instead, but I think this will work for now and we can always change it.